vsp.go

```diff
@@ -440,18 +704,188 @@ func (v *VSP) GetFeeAddress(ctx context.Context, ticketHash chainhash.Hash) (dcr
 	}
-	v.ticketToFeeLock.Unlock()
+	vsp.ticketToFeeLock.Unlock()
 
 	return feeAmount, nil
 }
 
+type valueOut struct {
+	Remember string
+	List     []string
+}
+
+func (mw *MultiWallet) GetVSPList(net string) {
+	var valueOut valueOut
+
+	mw.ReadUserConfigValue(VSPHostConfigKey, &valueOut)
+	var loadedVSP []*VSPInfo
+
+	for _, host := range valueOut.List {
+		info, err := mw.getInfo(host)
+		if err == nil {
+			loadedVSP = append(loadedVSP, &VSPInfo{
+				Host: host,
+				Info: info,
+			})
+		}
+	}
+
+	l, _ := getInitVSPInfo("https://api.decred.org/?c=vsp")
+	for h, v := range l {
+		if strings.Contains(net, v.Network) {
+			loadedVSP = append(loadedVSP, &VSPInfo{
+				Host: fmt.Sprintf("https://%s", h),
+				Info: v,
+			})
+		}
+	}
+
+	mw.VspList = loadedVSP
+}
+
+func (mw *MultiWallet) AddVSP(host, net string) error {
+	var valueOut valueOut
+
+	// check if host already exists
+	_ = mw.ReadUserConfigValue(VSPHostConfigKey, &valueOut)
+	for _, vsp := range valueOut.List {
+		if vsp == host {
+			return fmt.Errorf("existing host %s", host)
+		}
+	}
+
+	// validate host network
+	info, err := mw.getInfo(host)
+	if err != nil {
+		return err
+	}
+
+	if info.Network != net {
+		return fmt.Errorf("invalid net %s", info.Network)
+	}
+
+	valueOut.List = append(valueOut.List, host)
+	mw.SaveUserConfigValue(VSPHostConfigKey, valueOut)
+	mw.VspList = append(mw.VspList, &VSPInfo{
+		Host: host,
+		Info: info,
+	})
+
+	return nil
+}
+
+func (mw *MultiWallet) GetRememberVSP() string {
+	var valueOut valueOut
+
+	mw.ReadUserConfigValue(VSPHostConfigKey, &valueOut)
+	return valueOut.Remember
+}
+
+func (mw *MultiWallet) RememberVSP(host string) {
+	var valueOut valueOut
+
+	err := mw.ReadUserConfigValue(VSPHostConfigKey, &valueOut)
+	if err != nil {
+		log.Error(err.Error())
+	}
+
+	valueOut.Remember = host
+	mw.SaveUserConfigValue(VSPHostConfigKey, valueOut)
+}
+
+func (mw *MultiWallet) getInfo(host string) (*VspInfoResponse, error) {
+	sourceWallet := mw.WalletWithID(mw.OpenedWalletIDsRaw()[0])
+	if sourceWallet == nil {
+		return nil, fmt.Errorf("wallet doesn't exist")
+	}
+
+	ctx, cancel := mw.contextWithShutdownCancel()
+	mw.cancelFuncs = append(mw.cancelFuncs, cancel)
+
+	vsp := &VSP{}
+	vsp.client = newVSPClient(host, nil, sourceWallet.Internal())
+	vspInfo, err := vsp.GetInfo(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	return vspInfo, nil
+}
+
+// SetAutoTicketsBuyerConfig set ticket buyer configuration for a particular wallet
+func (wallet *Wallet) SetAutoTicketsBuyerConfig(vspHost string, purchaseAccount int32, amountToMaintain int64) {
+	wallet.SetLongConfigValueForKey(TicketBuyerATMConfigKey, amountToMaintain)
+	wallet.SetInt32ConfigValueForKey(TicketBuyerAccountConfigKey, purchaseAccount)
+	wallet.SetStringConfigValueForKey(TicketBuyerVSPHostConfigKey, vspHost)
+}
+
+// GetAutoTicketsBuyerConfig gets ticekt buyer config for a selected wallet
+func (wallet *Wallet) GetAutoTicketsBuyerConfig() *TicketBuyerConfig {
```

Comment:
`AutoTicketsBuyerConfig()` or just `TicketBuyerConfig()`, no need for the get prefix.